File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.java
class Solution {
    public int solution(int a, int b, int n) {
        int takeCoke = a;
        int receivedCoke = 0;
        int remainCoke = n;

        int answer = 0;
        while (remainCoke >= a) {
            takeCoke = remainCoke / a;
            receivedCoke = takeCoke * b;
            remainCoke = receivedCoke + (remainCoke % a);
            answer += receivedCoke;
        }
        return answer;
    }
}

File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.py
def solution(a, b, n):
    answer = 0
    
    while(n >= a):
        change_coke = (n // a) * b
        remain_coke = (n % a)
        
        answer += change_coke
        n = change_coke + remain_coke
        
    return answer